@kdlong 
the bug fix proposed here is not correct we are mixing Phi with Eta.

Step: DQM Spec: ['@nanogenDQM']
customising the process with customizeNanoGENFromMini from PhysicsTools/NanoAOD/nanogen_cff
Traceback (most recent call last):
  File "/cvmfs/cms-ib.cern.ch/nweek-02652/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-10-27-1100/bin/slc7_amd64_gcc820/cmsDriver.py", line 56, in <module>
    run()
  File "/cvmfs/cms-ib.cern.ch/nweek-02652/slc7_amd64_gcc820/cms/cmssw/CMSSW_11_2_X_2020-10-27-1100/bin/slc7_amd64_gcc820/cmsDriver.py", line 28, in run
    configBuilder.prepare()
  File "/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-10-27-2300/python/Configuration/Applications/ConfigBuilder.py", line 2273, in prepare
    self.pythonCfgCode += self.addCustomise()
  File "/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-10-27-2300/python/Configuration/Applications/ConfigBuilder.py", line 885, in addCustomise
    self.process=getattr(package,fcn)(self.process)
  File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-27-2300/python/PhysicsTools/NanoAOD/nanogen_cff.py", line 77, in customizeNanoGENFromMini
    nanoGenCommonCustomize(process)
  File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-27-2300/python/PhysicsTools/NanoAOD/nanogen_cff.py", line 56, in nanoGenCommonCustomize
    setGenEtaPrecision(process, CandVars.eta.precision)
  File "/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-27-2300/python/PhysicsTools/NanoAOD/nanogen_cff.py", line 137, in setGenEtaPrecision
    process.metMCTable.variables.eta.precision = precision
AttributeError: 'PSet' object has no attribute 'eta'

--- a/PhysicsTools/NanoAOD/python/nanogen_cff.py
+++ b/PhysicsTools/NanoAOD/python/nanogen_cff.py
@@ -6,7 +6,7 @@ from PhysicsTools.NanoAOD.genparticles_cff import *
 from PhysicsTools.NanoAOD.particlelevel_cff import *
 from PhysicsTools.NanoAOD.lheInfoTable_cfi import *
 from PhysicsTools.NanoAOD.genWeightsTable_cfi import *
-from PhysicsTools.NanoAOD.common_cff import CandVars
+from PhysicsTools.NanoAOD.common_cff import Var,CandVars
 
 nanoMetadata = cms.EDProducer("UniqueStringProducer",
     strings = cms.PSet(
@@ -51,8 +51,8 @@ def nanoGenCommonCustomize(process):
     process.particleLevel.lepMaxEta = 999.
     process.genJetFlavourTable.jetFlavourInfos = "genJetFlavourAssociation"
     # Same as default RECO
-    setGenPhiPrecision(process, CandVars.pt.precision)
-    setGenPtPrecision(process, CandVars.eta.precision)
+    setGenPtPrecision(process, CandVars.pt.precision)
+    setGenEtaPrecision(process, CandVars.eta.precision)
     setGenPhiPrecision(process, CandVars.phi.precision)
 
 def customizeNanoGENFromMini(process):
@@ -61,8 +61,9 @@ def customizeNanoGENFromMini(process):
     process.nanoAOD_step.insert(0, process.mergedGenParticles)
 
     process.metMCTable.src = "slimmedMETs"
-    process.metMCTable.variables.pt = Var("genMET.pt", float, doc="pt", precision=10)
-    process.metMCTable.variables.phi = Var("genMET.phi", float, doc="phi", precision=10)
+    process.metMCTable.variables.pt = Var("genMET.pt", float, doc="pt")
+    process.metMCTable.variables.phi = Var("genMET.phi", float, doc="phi")
+    process.metMCTable.variables.phi.precision = CandVars.phi.precision
 
     process.rivetProducerHTXS.HepMCCollection = "genParticles2HepMCHiggsVtx:unsmeared"
     process.genParticleTable.src = "prunedGenParticles"
@@ -79,8 +80,7 @@ def customizeNanoGENFromMini(process):
 
 def customizeNanoGEN(process):
     process.metMCTable.src = "genMetTrue"
-    process.metMCTable.variables.pt = Var("pt", float, doc="pt", precision=10)
-    process.metMCTable.variables.phi = Var("phi", float, doc="phi", precision=10)
+    process.metMCTable.variables = cms.PSet(PTVars)
 
     process.rivetProducerHTXS.HepMCCollection = "generatorSmeared"
     process.genParticleTable.src = "genParticles"
@@ -133,7 +133,6 @@ def setGenPtPrecision(process, precision):
 def setGenEtaPrecision(process, precision):
     process.genParticleTable.variables.eta.precision = precision
     process.genJetTable.variables.eta.precision = precision
-    process.metMCTable.variables.eta.precision = precision
     return process
 
 def setGenPhiPrecision(process, precision):

